middleware.go

```diff
@@ -139,7 +145,30 @@ func (tw traceware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		oteltrace.WithAttributes(semconv.HTTPServerAttributesFromHTTPRequest(tw.serverName, routePattern, r)...),
 		oteltrace.WithSpanKind(oteltrace.SpanKindServer),
 	)
-	defer span.End()
+	defer func() {
+		if recovered := recover(); recovered != nil {
+			defer panic(recovered)
```

Comment:
May I know, why `panic` called again in here? Is it intended?
At first, I thought this PR will handle if `panic` happens it won't crash the app.

Reply:
Hmm for me it should rethrow the panic, its job is just to record it, not to stop it.